app/assets/src/components/views/PipelineViz/PipelineViz.jsx

```diff
@@ -48,29 +50,98 @@ class PipelineViz extends React.Component {
       sidebarVisible: false,
       sidebarParams: {},
       hovered: false,
+      graphData: props.graphData,
     };
   }
 
   componentDidMount() {
     this.drawGraphs();
     window.addEventListener("resize", this.handleWindowResize);
+    // Only set update loop if the pipeline is still running.
+    if (!this.pipelineIsFinished()) {
+      this.updateLoop = setInterval(() => this.updateGraphs(), 1000);
```

Comment:
1 second seems excessive. 1 minute seems enough given pipeline speed.
Also, can you add this to constant or default props.